parse more bitmap tag and fix some bug

EvaWall opened the pull request.

EvaWall committed:
warn about parsing problem

EvaWall committed:
A language code of zero means no language.

EvaWall committed:
u_get() should return a number

EvaWall committed:
offload image data validating to image processing module

EvaWall committed:
handle DefineBitsJPEG3 tag

EvaWall committed:
handle DefineBitsLossless tag

EvaWall committed:
handle DefineBitsLossless2 tag

EvaWall committed:
handle DefineBitsJPEG4 tag

facundobatista reviewed (changes requested):
Hello! Thanks for this workl!!!!!!

Changes are beyond my reasoning of in which cases are needed, but I'm fine with that and eager to land it.

But note that tests fail. Some are simple (the change from None to 0, for example), but others are more complicated.

Please check that, thanks!

EvaWall committed:
update test for u_get()

EvaWall committed:
JPEG SOI marker is required for JPEGData

without SOI marker `im = PIL.Image.open(io.BytesIO(tag.JPEGData))`
will raise PIL.UnidentifiedImageError: cannot identify image file

EvaWall committed:
fix flake8 E501 line too long

EvaWall commented:
tests OK now.

the [commit about SHAPERECORDS](https://github.com/facundobatista/yaswfp/commit/ec4c9f83b882ba8743bf30e4108a6017deb09de6) works for some swf file, but not the test one. so removed from the pull request. will try to check it later.

thanks.

facundobatista reviewed (approved):
Awesome, thanks!!
